src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -466,6 +469,71 @@ RequestContext<object> requestContext
             await requestContext.SendResult(result.First().ToString());
         }
 
+        private async Task HandleGetAllCommandsRequest(
+            object param,
+            RequestContext<object> requestContext)
```

Comment:
If we're sending back a `List<PSCommandMessage>`, that should be the generic type of the `RequestContext` I think. Ideally we could return something where we can safely cache an empty result, like `PSCommandMessage[]`, so when `result` is `null` or empty, we can send back `Array.Empty<PSCommandResult>()` and save an allocation

Reply:
If I change this to use RequestContext<List<PSGetAllCommandsRequest>> it errors on line 127 where we specify the handler:

`Argument 2: cannot convert from 'method group' to 'Func<List<PSAllCommandsMessage>, RequestContext<object>, Task>' [PowerShellEditorServices.Protocol]`